lib/modules/manager/vendir/artifacts.spec.ts

```diff
@@ -0,0 +1,356 @@
+import { mockDeep } from 'jest-mock-extended';
+import { join } from 'upath';
+import { envMock, mockExecAll } from '../../../../test/exec-util';
+import { Fixtures } from '../../../../test/fixtures';
+import { env, fs, git, partial } from '../../../../test/util';
+import { GlobalConfig } from '../../../config/global';
+import type { RepoGlobalConfig } from '../../../config/types';
+import type { StatusResult } from '../../../util/git/types';
+import type { UpdateArtifactsConfig } from '../types';
+import * as vendir from '.';
+
+jest.mock('../../datasource', () => mockDeep());
+jest.mock('../../../util/exec/env');
+jest.mock('../../../util/http');
+jest.mock('../../../util/fs');
+jest.mock('../../../util/git');
+
+const adminConfig: RepoGlobalConfig = {
+  localDir: join('/tmp/github/some/repo'), // `join` fixes Windows CI
+  cacheDir: join('/tmp/renovate/cache'),
+  containerbaseDir: join('/tmp/renovate/cache/containerbase'),
+};
+
+const config: UpdateArtifactsConfig = {};
+const vendirLockFile1 = Fixtures.get('vendir_1.lock');
+const vendirLockFile2 = Fixtures.get('vendir_2.lock');
+const vendirFile = Fixtures.get('vendir.yml');
+
+describe('modules/manager/vendir/artifacts', () => {
+  beforeEach(() => {
+    env.getChildProcessEnv.mockReturnValue(envMock.basic);
+    GlobalConfig.set(adminConfig);
+  });
+
+  afterEach(() => {
+    GlobalConfig.reset();
+  });
+
+  it('returns null if no vendir.lock.yml found', async () => {
+    const updatedDeps = [{ depName: 'dep1' }];
+    expect(
+      await vendir.updateArtifacts({
+        packageFileName: 'vendir.yml',
+        updatedDeps,
+        newPackageFileContent: '',
+        config,
+      })
+    ).toBeNull();
+  });
+
+  it('returns null if updatedDeps is empty', async () => {
+    expect(
+      await vendir.updateArtifacts({
+        packageFileName: 'vendir.lock.yml',
+        updatedDeps: [],
+        newPackageFileContent: '',
+        config,
+      })
+    ).toBeNull();
+  });
+
+  it('returns null if unchanged', async () => {
+    fs.readLocalFile.mockResolvedValueOnce(vendirLockFile1 as any);
+    fs.getSiblingFileName.mockReturnValueOnce('vendir.lock.yml');
+    const execSnapshots = mockExecAll();
+    fs.readLocalFile.mockResolvedValueOnce(vendirLockFile1 as any);
+    fs.privateCacheDir.mockReturnValue(
+      '/tmp/renovate/cache/__renovate-private-cache'
+    );
+    fs.getParentDir.mockReturnValue('');
+    const updatedDeps = [{ depName: 'dep1' }];
+    expect(
+      await vendir.updateArtifacts({
+        packageFileName: 'vendir.yml',
+        updatedDeps,
+        newPackageFileContent: vendirFile,
+        config,
+      })
+    ).toBeNull();
+    expect(execSnapshots).toMatchSnapshot();
```

Comment:
no snapshot

Reply:
I'm not sure how to do this without snapshots.

We are not trying to test vendir specifically, but rather that the TS code works properly. When I remove the the snapshot call vendir fails because the lockfile doesn't actually exist on the filesystem. Is there an example you can point me too? Because I copied this from the poetry manager and they use snapshots?

Reply:
Ah, I figured it out, you meant have a snapshot passed. Not don't use snapshots